tests/apis.js

```diff
@@ -52,80 +51,300 @@ test.cb.after.always(t => {
     });
 });
 
+/**
+ * Tests attributes with references
+ * @param  {object} t - ava testing
+ * @param  {object|array} attrs object containing attributes to test
+ */
+const referencerTests = (t, attrs) => {
+  if (typeof attrs !== 'object' || attrs === undefined) {
+    return;
+  }
+
+  if (attrs.hasOwnProperty('attributes')) {
+    referencerTests(t, attrs.attributes);
+
+    return;
+  }
+
+  if (Array.isArray(attrs)) {
+    attrs.forEach(attr => {
+      Object.keys(attr).forEach(atr => {
+        if (attr[atr].hasOwnProperty('attributes')) {
+          referencerTests(t, attr[atr].attributes);
+
+          return;
+        }
+        else if (attr[atr].hasOwnProperty('id')) {
+          t.true(attr[atr].hasOwnProperty('meta'), 'attribute in array contains meta');
+
+          return;
+        }
+      });
+    });
+  }
+
+  // check if this object is an attribute object
+  if (Object.keys(attrs).indexOf('id') !== -1) {
+    t.true(attrs.poo.hasOwnProperty('meta'), 'attribute in array contains meta');
+
+    return;
+  }
+
+  Object.keys(attrs).forEach(attr => {
+    if (attr.split('-').indexOf('referencer') > -1) {
+      const item = attrs[attr];
+
+      // if item has attributes
+      if (item.hasOwnProperty('attributes')) {
+        referencerTests(t, item.attributes);
+
+        return;
+      }
+
+      // if it's an array, recurseit!
+      if (Array.isArray(item)) {
+        referencerTests(t, item);
+
+        return;
+      }
+
+      if (typeof item === 'object' && typeof item[Object.keys(item)[0]] === 'object' && item[Object.keys(item)[0]].hasOwnProperty('attributes')) {
+        referencerTests(t, item.attributes);
+
+        return;
+      }
+
+      if (Object.keys(item).indexOf('id') === -1 && Object.keys(item).length >= 1) {
+        Object.keys(item).forEach(itm => {
+          t.true(item[itm].hasOwnProperty('meta'), 'attribute in array contains meta');
+
+          return;
+        });
+      }
+
+      // non-repeatable; single input
+      else {
+        t.true(item.hasOwnProperty('meta'), 'attribute in array contains meta');
+
+        return;
+      }
+    }
+  });
+};
+
 //////////////////////////////
 // Utils - attributes
 //////////////////////////////
-test('Utils: attributes', t => {
-  const item = Math.round(Math.random() * (content.length - 1));
-  let expected = content[item];
+test.serial.skip('Utils: attributes - empty query', t => {
+  const random = Math.round(Math.random() * live.length - 1);
```

Comment:
This `random` was right the first time. `Math.round(Math.random() * (live.length - 1)). Need the parenthesis around the `-1` because of PEMDAS